pkgs/by-name/mc/mcap-cli/package.nix

```
@@ -39,6 +44,13 @@ buildGoModule {
     "-skip=TestCat|TestInfo"
   ];
 
+  postInstall = ''
+    installShellCompletion --cmd mcap \
+      --bash <($out/bin/mcap completion bash) \
+      --fish <($out/bin/mcap completion fish) \
+      --zsh <($out/bin/mcap completion zsh)
+  '';
```

This doesn't work when cross compiling (e.g. building `pkgsCross.aarch64-multiplatform.mcap-cli` on `x86_64-linux`). In this case, you could either disable the generation of shell completions or use an emulator (if available):

```suggestion
  postInstall = lib.optionalString (stdenv.hostPlatform.emulatorAvailable buildPackages) (
    let
      emulator = stdenv.hostPlatform.emulator buildPackages;
    in
    ''
      installShellCompletion --cmd mcap \
        --bash <(${emulator} $out/bin/mcap completion bash) \
        --fish <(${emulator} $out/bin/mcap completion fish) \
        --zsh <(${emulator} $out/bin/mcap completion zsh)
    ''
  );
```

---

Thanks for review. Fixed.